docs/modules/integrate/pages/vertx-step-by-step-buide.adoc

```diff
@@ -0,0 +1,477 @@
+====
+TODO (should be removed when fixed before merging)
+
+There seems tobe a bug in 5.0.0-SNAPSHOT
+change
+
+`<launcher.class>io.vertx.core.Launcher</launcher.class>`
+
+to
+
+`<launcher.class>io.vertx.launcher.application.VertxApplication</launcher.class>`
+
+END TODO
+====
+
+== Storing Data in Session
+
+Go to the `MainVerticle.java` file and replace the contents of the start method with the following:
+
+NOTE: This guide uses 2 space indentation, which is customary for Vertx projects due to high number of nested callbacks.
+
+[source,java]
+----
+  public void start() {
+    // Create a Router
+    Router router = router(vertx);
+
+    // Create local SessionStore
+    SessionStore store = LocalSessionStore.create(vertx);
+
+    // Use the SessionStore to handle all requests
+    router.route()
+      .handler(SessionHandler.create(store));
+
+    router.route(HttpMethod.PUT, "/").handler(context -> {
+      context.request().bodyHandler(body -> {
+        List<String> messages = getMessagesFromSession(context);
+
+        JsonObject json = body.toJsonObject();
+        String message = json.getString("message");
+        messages.add(message);
+
+        putMessagesToSession(context, messages);
+
+        context.json(
+          new JsonObject()
+            .put("messages", messages)
+        );
+      });
+    });
+
+    // Create the HTTP server
+    vertx.createHttpServer()
+      // Handle every request using the router
+      .requestHandler(router)
+      // Start listening
+      .listen(8888)
+      // Print the port
+      .onSuccess(server ->
+        System.out.println(
+          "HTTP server started on port " + server.actualPort()
+        )
+      );
+  }
+
+  private static List<String> getMessagesFromSession(RoutingContext context) {
+    String messages = context.session().get("messages");
+    if (messages == null) {
+      return new ArrayList<>();
+    } else {
+      return new ArrayList<>(Arrays.asList(messages.split(",")));
+    }
+  }
+
+  private void putMessagesToSession(RoutingContext context, List<String> messages) {
+    context.session().put("messages", String.join(",", messages));
+  }
+----
+
+[source,bash]
+----
+$ http put localhost:8888 message=Hello\ World!
+HTTP/1.1 200 OK
+content-length: 29
+content-type: application/json
+set-cookie: vertx-web.session=ed22f77473a7f613c9305431a62832a6; Path=/
+
+{
+    "messages": [
+        "Hello World!"
+    ]
+}
+
+----
+
+Execute another request with the cookie.
+
+[source,bash]
+----
+$ http put localhost:8888 'Cookie:vertx-web.session=ed22f77473a7f613c9305431a62832a6' message=Hello\ World\ 2!
+HTTP/1.1 200 OK
+content-length: 46
+content-type: application/json
+
+{
+    "messages": [
+        "Hello World!",
+        "Hello World 2!"
+    ]
+}
+
+----
+
+== Distributed Sessions
+
+Let's modify the code, so we can start multiple instances easily - the application will start on defined port, and when the port is not available it will search for another port:
```

Comment:
```suggestion
Let's modify the code, so we can start multiple instances easily - the application will start on the defined port, and when the port is not available it will search for another port:
```